fdbclient/NativeAPI.actor.cpp

```diff
@@ -3373,3 +3373,56 @@ ACTOR Future<Void> snapCreate(Database cx, Standalone<StringRef> snapCmd, UID sn
 		throw;
 	}
 }
+
+ACTOR Future<bool> checkSafeExclusions(Database cx, vector<AddressExclusion> exclusions) {
+	TraceEvent("ExclusionSafetyCheckBegin")
+	    .detail("NumExclusion", exclusions.size())
+	    .detail("Exclusions", describe(exclusions));
+	state ExclusionSafetyCheckRequest req(exclusions);
+	state bool ddCheck;
+	loop {
+		choose {
+			when(wait(cx->onMasterProxiesChanged())) {}
+			when(ExclusionSafetyCheckReply _ddCheck = wait(loadBalance(
+			         cx->getMasterProxies(false), &MasterProxyInterface::exclusionSafetyCheckReq, req, cx->taskID))) {
+				ddCheck = _ddCheck.safe;
+				break;
+			}
+		}
+	}
+	TraceEvent("ExclusionSafetyCheckCoordinators");
+	state ClientCoordinators coordinatorList(cx->getConnectionFile());
+	state vector<Future<Optional<LeaderInfo>>> leaderServers;
+	for (int i = 0; i < coordinatorList.clientLeaderServers.size(); i++) {
```

Comment:
Under which scenario, we want to mark a coordinator as failed, instead of letting coordinators heal by itself?

Reply:
It is unclear if this has an actual use case, but since regular excludes allow the exclusion of coordinators, I figured it would be safe to have it checked for here as well

Reply:
I remembered the case Markus mentioned before about the problem caused by excluding limping coordinators. ok, then.

Reply:
In the test below, we may want to explicitly test excluding a coordinator by BUGGIFY it.